main.py

```diff
@@ -0,0 +1,73 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+    skip: int | None = 0,
+    limit: int | None = 5,
+    db: Session = Depends(get_db)):
+    return crud.get_all_authors(db, skip, limit)
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_author_by_id(
+    author_id: int,
+    db: Session = Depends(get_db),
+):
+    return crud.get_author(db, author_id)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author_by_name(db, author.name)
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Such author with name '{author.name}' already exists."
+        )
+
+    return crud.create_author(db, author)
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books(
+    skip: int | None = 0,
```

Comment:
The type hint `int | None` is not compatible with Python versions below 3.10. If you are using an earlier version, consider using `Optional[int]` from the `typing` module instead.